pkg/compute/store/boltdb/store.go

```diff
@@ -404,35 +444,64 @@ func (s *Store) deleteExecution(tx *bolt.Tx, executionID string) error {
 	}
 
 	// Delete the bucket with the execution-id within the history bucket
-	historyBucket := tx.Bucket([]byte(BucketHistoryName))
+	historyBucket := s.getHistoryBucket(tx)
 	err = historyBucket.DeleteBucket([]byte(executionID))
 	if err != nil {
-		errorList = errors.Join(errorList, err)
+		return err
 	}
 
-	return errorList
+	return nil
 }
 
 // Close ensures the database is closed cleanly
 func (s *Store) Close(ctx context.Context) error {
 	return s.database.Close()
 }
 
-func (s *Store) GetExecutionCount(ctx context.Context) (uint, error) {
+func (s *Store) GetExecutionCount(ctx context.Context, state store.ExecutionState) (uint, error) {
 	log.Ctx(ctx).Trace().
 		Msg("boltdb.GetExecutionCount")
 
-	// TODO(ross) So, counting based on execution state .... seems like
-	// maintaining a prefix for a counter is excessive
-
-	// var counter uint
-	// for _, execution := range s.executionMap {
-	// 	if execution.State == store.ExecutionStateCompleted {
-	// 		counter++
-	// 	}
-	// }
-	// return counter, nil
-	return 0, nil
+	// We have to wait here to ensure the counter has been populated,
+	// so we will wait until we know it has finished
+	s.starting.Wait()
+	return uint(s.stateCounter.Get(state)), nil
```

Comment:
does it make sense to change `GetExecutionCount` to return uint64 instead of down casting?